### ISSUE TITLE
Admin\Operator - External User can cancel an operator access request

### ISSUE THREAD

**Issue by dleard** [bcgov/cas-registration#364]
#### Description:

An IO user should be able to cancel an access request. On the request-access/request page in the designs there is a `cancel request` button. When clicked, this button should cancel (delete the user_operator entry) the access request for that user-operator pair.

[Wireframe](https://www.figma.com/file/LsDmLDCdnJqI8UahyqvOD8/OBPS-App-Designs?type=design&node-id=2621%3A3889&mode=design&t=0sO2L1UujkO2DS3j-1)

#### Acceptance Criteria:

Given I have requested access to an operator
And I am on the received page
When I click `cancel request`
Then a confirmation modal appears asking me to confirm the cancellation. 

<img width="589" alt="Image" src="https://github.com/user-attachments/assets/c78081c7-7574-4e0f-9cb6-c4193715db55" />

Given I confirm the cancellation,
Then my access request is cancelled

Given I cancel,
Then the confirmation modal closes and I am redirected back to the select operator page.

#### Development Checklist:

- [x] The received page has a `cancel request` button
- [x] When clicked, the button deletes (hard deletes) the corresponding row from the user_operator table
- [ ] ~After a request has been cancelled the user should see a confirmation that their request was cancelled~
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [x] User story is included
- [x] User role and type are identified
- [x] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by patrickisaac** [comment]
Refinement Notes:
- Potential to tackle for GO LIVE if there's time, but can be de-prioritized

**Comment by zoeyli-46** [comment]
Potential additions: After cancelling, they can request access to the same operator again or to a different operator.

**Comment by Sepehr-Sobhani** [comment]
@zoeyli-46 I suggest redirecting the user to the `Select Operator` page after canceling the previous request, as staying on this page no longer makes sense for them. What do you think?

**Comment by zoeyli-46** [comment]
@sepehr yes! You are right - it would be better to redirect the user to the select operator page again. Revised the AC

**Comment by Sepehr-Sobhani** [comment]
Thank you, @zoeyli-46 ! One more thing: We typically show confirmation messages, like a snackbar, when the user stays on the same page. However, in this case, we're redirecting the user to another page after deleting the access request. While it's possible to show a snackbar even after a redirect, it would require a separate card and integrating the snackbar across the entire app, which is quite a bit of work. If this is an essential requirement, I can create a separate card to implement it.

**Comment by zoeyli-46** [comment]
Hi @Sepehr-Sobhani it is not a must have. It's still clear to the user that their request has been cancelled if they get redirected back to the previous page. I will remove the snackbar from the AC.

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document for reference](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/EZg6czWUaSpDv9l4zImldnkBGot_dex5gZGj9miKpYeyzw?e=i9mXc6).

I can verify that:
- [x] `Cancel Request` button is visible in the received page
- [x] Confirmation modal is displayed upon clicking `Cancel Request`
- [x] When user clicks on `No, Don't Cancel` the confirmation modal disappears, page stays at the received page, and the request is still visible as Pending 
- [x] When user clicks on `Yes, Cancel This Request`, page reroutes to Select Operator where user can select another operator, and user access request is not visible and is deleted from the user_operator table
- [x] When administrator/internal user approves request, user has access to the operator

Moving to Pending PO for review of test. Thank you!

**Comment by vesselak** [comment]
Reviewed the QA testing record and accepting ticket.
